lib/routes/seagullApi.js

```diff
@@ -348,7 +348,10 @@ module.exports = function (crudHandler, userApiClient, gatekeeperClient, metrics
           return next(false);
         }
 
-        if (trustorUserPermissions && !_.isUndefined(trustorUserPermissions[req.params.userid])) {
+        // Check to see if the user has trustor permissions for the requested user ID
+        // or `hydrophone` (which is listed as a trustor when claiming a VCA-created custodial account).
+        var trustorUserIds = _.keys(trustorUserPermissions);
+        if (!_.isEmpty(_.intersection(trustorUserIds, [req.params.userid, 'hydrophone']))) {
```

Comment:
@cbwebdevelopment I'm not sure this is the right way to solve this problem. I'm going to have to look into the original changes that @pazaan made and figure out the best way to handle this.

We may want to temporarily rollback seagull so we allow custodian accounts to be claimed.

Reply:
Okay, now I'm confused. Let me look more into this. I may be thinking of two separate things.

Reply:
Ah, I see the problem now. That changes that @pazaan put in place failed to account for some requests that come directly from another service implicitly have "server" privilege and therefore should not do a user check. My bad, as I missed it when I did my review.

Unfortunately, the solution is a bit complicated. We have to pull out the logic of `crudHandler.getDoc` because there are two flows that need to happen. If the requestor is a server (`req._tokendata.isserver`) then immediately call `crudHandler.getDoc`. If not, then need to call the `gatekeeperClient.groupsForUser` and within its callback, do the current check for `trustorUserPermissions` and `if (hasTrustorPermissions || collection === 'profile')` then call `crudHandler.getDoc`.

Does that make sense? If not, either we can chat about it or I can make the change. Let me know.

Also, we never compare the `req.params.userid` against a server name. This is fragile and could easily break. If it informational only. We should only write code against `req._tokendata.isserver`.